### ISSUE TITLE
Links with parenthesis (like wikipedia links) are not correct when using autolinks

### ISSUE THREAD

**Issue by ccampbell** [russross/blackfriday#291]
For example if you have some markdown text with:

```md
Check out this link:
https://en.wikipedia.org/wiki/St._Lucia_(musician)
```

The final markup ends up being

```html
<p>Check out this link:</p>

<p><a href="https://en.wikipedia.org/wiki/St._Lucia_(musician" rel="nofollow">https://en.wikipedia.org/wiki/St._Lucia_(musician</a>)</p>
```

Notice the final closing parenthesis is not matched

**Comment by rtfb** [comment]
Upon reading this I was certain this is a duplicate, but after carefully reviewing the list of issues twice, I can't find the other report, weird.... Wherever the original is hiding, thanks for reporting!

**Comment by ccampbell** [comment]
I tried to find another ticket first, but I couldn't find one. I did find this in the tests:

https://github.com/russross/blackfriday/blob/93622da34e54fb6529bfb7c57e710f37a8d9cbd8/inline_test.go#L571

Looks like there are tests for when you format the link yourself, but not the autolink part

**Comment by shurcooL** [comment]
@rtfb, are these not it?

https://github.com/russross/blackfriday/issues/123
https://github.com/russross/blackfriday/issues/116

**Comment by rtfb** [comment]
Yup, they are. Didn't expect them to be fixed, so only looked among the open ones.

**Comment by ccampbell** [comment]
@rtfb those tickets are for when you manually put in the markdown for the URL. That **is** fixed and working correctly. 

This ticket is for if you put the URL in your markdown without wrapping it in `[]()`

**Comment by cup** [comment]
https://github.com/russross/blackfriday/issues/116
https://github.com/russross/blackfriday/issues/123

https://github.com/russross/blackfriday/issues/291
https://github.com/russross/blackfriday/pull/222
https://github.com/russross/blackfriday/pull/356
https://github.com/russross/blackfriday/pull/358

**Comment by guregu** [comment]
For anyone reading this who is desperate for a workaround, you can use goquery to "fix" the links.

```go
doc.Find("a[href*='(']:not([href$=')'])").Each(func(_ int, sel *goquery.Selection) {
	if len(sel.Nodes) == 0 {
		return
	}
	sibling := sel.Nodes[0].NextSibling
	if sibling == nil || sibling.Type != html.TextNode || !strings.HasPrefix(sibling.Data, ")") {
		return
	}
	href, _ := sel.Attr("href")
	fixed := href + ")"
	sel.SetAttr("href", fixed)
	if sel.Text() == href {
		sel.SetText(fixed)
		sibling.Data = strings.TrimPrefix(sibling.Data, ")")
	}
})
```
